### ISSUE TITLE
SIF Enhancement - Features to Manage Director Appeals (LP) (Ready for UAT on Prod)

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#12918]
### Create new SIF folder process to manage appeals made to the TPW Director
_Identified need during meeting on 7/18/23 with Nathan A & Dan R._

**Acceptance Criteria** 

- A new Process needs to be added to the SIF folder. 
- Process Description - SIF Director Appeal
- Process Group: SIF Appeals
    - `Create a new process group for SIF Appeals` 
- Process Insertion - Inserted on folder creation 
- Default Assigned User - None
- Default To Start and To End Date - None
- Default Process Status - Open
- A new Process info needs to be added to the SIF Director Appeal Process
- Process Info Description:  SIF Appeal Related PR#
- Process Info type: Alphanumeric
- Process Attempts - Appeal Packet Received, Appeal Packet Incomplete, Appeal Packet Complete, Director Determination Denied, Director Determination Approved
- The process attempts should perform the following post-events when added to the process:

**Appeal Packet Received**
- Pre-Condition: If Applicant People record does not have an email ID, system should allow the user to save this attempt but display a Warning message "Applicant does not have an email address, so an email will not be sent as part of this attempt"
- Pre-Condition (hard stop): Process Info SIF Appeal Related PR# must be populated
- Pre-Condition (hard stop): If this is not the 1st attempt result, the previous attempt must be 'Appeal Packet Incomplete'
- Update Folder Status to 'Pending Appeal Packet Review'
- Send email to People Type 'Applicant' `content TBD`
- Update Assigned User to 'ATD SIF Reviewer' user
- Set Process Status to 'Appeal Packet Received'
- Set To Start Date as SysDate (Current Date) 
- Set To End Date as SysDate + 3 Calendar weeks (21 days)
- Remove Ended Date in the process. 

**Appeal Packet Incomplete** 
- Pre-Condition (hard stop if not true): the previous attempt saved is 'Appeal Packet Received.' 
- Pre-Condition (hard stop) - Do not save if attempt comment is null
- Pre-Condition: If Applicant People record does not have an email ID, system should allow the user to save this attempt but display a Warning message "Applicant does not have an email address, so an email will not be sent as part of this attempt"
- Pre-Condition (hard stop): Process Info SIF Appeal Related PR# must be populated
- Update Folder Status to 'Appeal Packet Incomplete'
- Set Process Status to 'Appeal Packet Incomplete'
- Send email to People Type 'Applicant' with Attempt Comments from this attempt. `Content TBD`
- Insert Ended Date in the process as SysDate (Current Date).

**Appeal Packet Complete** 
- Pre-Condition (hard stop if not true): the previous attempt saved is 'Appeal Packet Received.'
- Pre-Condition: If Applicant People record does not have an email ID, system should allow the user to save this attempt but display a Warning message "Applicant does not have an email address, so an email will not be sent as part of this attempt"
- Pre-Condition (hard stop): Process Info SIF Appeal Related PR# must be populated
- Update Folder Status to 'Pending Director Decision'
- Set Process Status to 'Pending Director Decision'
- Send email to People Type 'Applicant'. `Content TBD`
- Remove Ended Date in the process 
- Remove To End Date in the process
- Update To Start Date as SysDate (Current Date).

**Director Determination - Denied** 
- Update Folder Status to 'In Review'
- Set Process Status to 'Closed'
- Do not send system email
- Insert Ended Date in the process as SysDate (Current Date).
- Insert another SIF Appeals Process in the folder with no assigned user and no To start and To End Dates and with process status 'Open'.

**Director Determination - Approved** 
- Pre-Condition: If Applicant People record does not have an email ID, system should allow the user to save this attempt but display a Warning message "Applicant does not have an email address, so an email will not be sent as part of this attempt"
- Pre-Condition (hard stop): Process Info SIF Appeal Related PR# must be populated
- Update Folder Status to 'In Review'
- Send email to People Type 'Applicant' `Content TBD`
- Set Process Status to 'Closed'
- Insert Ended Date in the process as SysDate (Current Date).
- Insert another SIF Appeals Process in the folder with no assigned user and no To start and To End Dates and with process status 'Open'.

**Comment by TracyLinder** [comment]
### Scoping Notes

**Q:** What is considered an appeal? --- 
- Formal process must be followed. Applicant must submit a Finding of Fact with 1/6 defined criteria identified. Then SIF staff is required to do certain things; no hard deadlines but must be done in order. 
- There is a defined workflow with defined criteria.  

How is an appeal different from a refund? 
- An appeal is an assertion was made in assessing the SIF fee, and the appeal is approved. The outcome is a refund. 

Are all refunds an outcome of an appeal? 
- No. There's the 10 year refund. Sometimes there are mistakes identified that don't need an appeal as staff agrees at the division level to refund monies already paid, also situations we can't predict.

**Comment by TracyLinder** [comment]
[Appeals Packet_Finding of Fact.docx](https://github.com/cityofaustin/atd-data-tech/files/12306818/Appeals.Packet_Finding.of.Fact.docx)

**Comment by TracyLinder** [comment]
### New Process - SIF Appeals

Description: SIF Appeals


**Process Attempts**

Appeal Packet Received & Returned 
Appeal Packet Received & Accepted
Director Meeting Scheduled
Director Meeting
Director Determination - Approved
Director Determination - Denied



Process Infos
Process Checklist
Defined Process Attachments

**Comment by TracyLinder** [comment]
### Next scoping meeting scheduled Tue 8/15/2023 3:00 PM - 4:00 PM

**Comment by TracyLinder** [comment]
### 2nd Scoping Mtg 8/18/23

- We created a miro workflow, [here](https://miro.com/app/board/uXjVMtcUaZ0=/?moveToWidget=3458764561873833852&cot=14).
- @vschandramouli will translate the workflow into functional criteria for the developer to build with. 
- @vschandramouli I'd like to look at the AC before it goes to the backlog please.

**Comment by TracyLinder** [comment]
### Next Scoping Meeting is 9/6/23
- We will review the acceptance criteria
- @vschandramouli FYI to be ready by this meeting. Thank you.

**Comment by vschandramouli** [comment]
Outstanding questions:

1. Email body and subject text 
2. Should we validate that Appeal Packet received attempt is made before Appeal Packet decision attempt is made?
3. Should we validate that Appeal Packet approved attempt has been made before director determination attempts are being made?

**Comment by TracyLinder** [comment]
### Final AC Review
_Aber, Nathan, Tracy, Vish_

1. Validations should be made on 2 attempts: Appeal Packet Incomplete and Appeal Packet Complete

**Comment by TracyLinder** [comment]
@vschandramouli I'm done updating the AC. Could you please review, make corrections, and move to the backlog? 

@Nadin-Nader This can go On Deck in the next sprint beginning 9/20, or sooner at your discretion. It's a new SIF folder process with functional attempts. It will probably go through 2-3 iterations.

**Comment by vschandramouli** [comment]
@TracyLinder @Nadin-Nader 

Tracy, I removed the following line from Director Determination attempts: 
- Validation: the previous attempt saved is 'Appeal Packet Received.'

I seem to remember us pitching this validation to the stakeholders, but they wanted to only keep it to the appeal packet attempts, not the director determination attempts. Please correct me if I am wrong. 

Also to note, the email body content is TBD. Should that be made another ticket?

Reviewed AC and made changes, ready for backlog.

**Comment by TracyLinder** [comment]
Vish's AC updates are correct. I think we can begin development without the email content. 
@vschandramouli can you please create a new ticket for the email content of this work? Add this issue as a blocker.

**Comment by TracyLinder** [comment]
@Nadin-Nader I'd like to get this on deck for the next sprint beginning 9/20 if possible.

**Comment by Nadin-Nader** [comment]
For:
Appeal Packet Incomplete

Pre-Condition (hard stop if not true): the previous attempt saved is 'Appeal Packet Received.'

Message will be anything that makes the user use the previous attempt made

**Comment by Nadin-Nader** [comment]
Workflow Screenshot:
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNVphQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a2884ac0acef139e0d16196743395f4dba6af290/image.png)

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Development and testing is in progress

**Comment by TracyLinder** [comment]
Sree will demo to TDS today at the existing meeting to look at the appeals email

**Comment by TracyLinder** [comment]
### 11/7/23 Sync with team & customers

- Need a validation on all attempts with an email to be sure the applicant record has a valid email listed. 
- Need a new process info **SIF Appeal Related PR#**

**Comment by vschandramouli** [comment]
Updated AC @Sreelatha-BlazeKannan @TracyLinder

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Localhost to DEV commit submitted in Github desktop; However the Phabricator issue is not fixed yet. so not able to submit DP in Phabricator for it to be audited. Waiting on CTM To fix the Phabricator issue.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan this has been sent to the DevOps CM Team to be deployed to DEV under **SCTASK0371412**

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Tested in DEV and its working as expected. submitted TEST DP for your reference. Kindly approve and create task id for the same.

Commit ID: e72965c6a23d9ef62a506de747267179619d10d9

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan this has been sent to the DevOps CM Team to be deployed to TEST under **SCTASK0372200**

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Tested in TEST Environment; all looks good and ready for TEST UAT

**Comment by TracyLinder** [comment]
### Tracy's Test Comments:
All of the AC looks to be working correctly to me! 🎉 I do want to request an update to the AC, see below. 
FYI @Sreelatha-BlazeKannan and @vschandramouli I will update the AC description also. 

- Create a new process group called **SIF Appeals**
- Update the process group code to the new SIF Appeals group.

**Comment by Nadin-Nader** [comment]
Update:
Working new AC changes

**Comment by TracyLinder** [comment]
@Sreelatha-BlazeKannan please update this when you can.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Along with #14377 these changes are submitted to DEV. I will be testing the same today.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Changes are deployed to TEST; Tested the same and looks good; Ready for TEST UAT.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - AC Change with New Process Group appears as below now.

![image](https://github.com/cityofaustin/atd-data-tech/assets/44907715/adb4cdba-ce31-4bf5-8fc4-3f61f8aa0cd8)

**Comment by TracyLinder** [comment]
Appeal Packet Received - only save if it's the first attempt, or if the last saved (most recent) attempt is Appeal Packet Incomplete.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan this has been sent to the DevOps CM Team to be deployed to DEV and TEST under **SCTASK0391455**

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - This is tested in both DEV & TEST with AC Changes for Incomplete conditions/validations.
All working as expected.
DEV FolderRSN - 13203141
TEST FolderRSNs - 13206341, 13206343

**DRAFT PROD DP’s of #12918 & #14377 sent. There will be a conflict that’s why these 2 cards commit must be done together to get all our AC changes.**

**Please find the Commit ID’s and its Order for promotion.**
![image](https://github.com/cityofaustin/atd-data-tech/assets/44907715/d788cd5a-2697-4ad5-b240-a3d77f1b8833)

**Comment by Nadin-Nader** [comment]
Sent an  email to the TDS team asking for this to be UATd on TEST. Waiting for response

**Comment by TracyLinder** [comment]
### Tracy's Testing

There are many bugs to address with the attempt error messaging and validations.

**Comment by TracyLinder** [comment]
### Testing Update
- The bugs i ran into worked themselves out somehow
- I asked TDS to test

**Comment by TracyLinder** [comment]
### UAT ✅
- Tested with Nathan A. today and have approval to move to PROD
- FYI @Lpriya-LIT

**Comment by Lpriya-LIT** [comment]
@Nadin-Nader Created prod deployment request , SCTASK0411259

**Comment by Lpriya-LIT** [comment]
@TracyLinder @Nadin-Nader Changes deployed to Prod , Ready for UAT

**Comment by Nadin-Nader** [comment]
@Lpriya-LIT i think this may have to go to CTM but can you please verify and let me know?

**Comment by TracyLinder** [comment]
@Nadin-Nader and @Lpriya-LIT can you please take a look at this one and get it moving forward again? Thank you.

**Comment by Nadin-Nader** [comment]
Update:
The procedure is not compiling right on PROD. This was brought up to CTM wo recompiled the procedure code on PROD to no avail. This issue is not happening on TEST so priya will need to troubleshoot and compare TEST to PROD.

**Comment by TracyLinder** [comment]
@Nadin-Nader as discussed, can you please send this to CTM for assistance?

**Comment by TracyLinder** [comment]
Hi @Nadin-Nader can you please provide an update on this ticket? 
I'm putting it back on deck. We need to get this closed out.

**Comment by Lpriya-LIT** [comment]
Update - Compared the procedure "Attempt_SIF_00084105" and "DefaultAttempt_SIF_00084105" in Test and Prod . There is no difference .
On test , it works as expected but on Prod we get DefaultAttempt_SIF_00084105 is invalid error.
In general Invalid procedure error is a compilation issue.
If possible , we shall schedule a call with ctm to compile these procedures both on Amanda admin and prod database.

**Comment by Lpriya-LIT** [comment]
Update - The procedure " DefaultAttempt_SIF_00084105" has been recompiled, @Nadin-Nader and I tried to do a attempt on Prod we didnt get the invalid procedure error now. 
@TracyLinder Could you please verify whether it is working as expected.

**Comment by Nadin-Nader** [comment]
@TracyLinder can you verify if this attempt is giving the Invalid Procedure error?

**Comment by TracyLinder** [comment]
### PROD Testing Approved

@Lpriya-LIT @Nadin-Nader I did extensive testing in prod and found no issues. 🎉
I will inform the SIF team and close this issue.
